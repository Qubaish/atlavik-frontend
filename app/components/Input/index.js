import React from 'react';
import PropTypes from 'prop-types';

function Input({val, onChange}) {
    return (
        <div className="row">
            <input
                placeholder="Name"
                id="name"
                type="text"
                name="name"
                onChange={onChange}
                value={val}
            />
        </div>
    )
}

Input.propTypes = {
    onChange: PropTypes.func,
    val: PropTypes.string,
};

export default Input;
