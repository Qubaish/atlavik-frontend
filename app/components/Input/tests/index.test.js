import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Input from '../index';

describe('<Input />', () => {
    it('should render a Input', () => {
        const container = shallow(
            <Input val="Dubai" onChange={jest.fn()} />
        );
        expect(shallowToJson(container)).toMatchSnapshot();
    });

    it('should called onChange', () => {
        const submitSpy = jest.fn();
        const container = shallow(
            <Input val="Dubai" onChange={submitSpy} />
        );
        container.find('input').props().onChange();
        expect(submitSpy).toHaveBeenCalled();
    });
});
