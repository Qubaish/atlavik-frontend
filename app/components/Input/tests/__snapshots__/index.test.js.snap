// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render a Input 1`] = `
<div
  className="row"
>
  <input
    id="name"
    name="name"
    onChange={[MockFunction]}
    placeholder="Name"
    type="text"
    value="Dubai"
  />
</div>
`;
